File: tests/python/cases/case_022_stage_order.py
"""
Task case 022: stage order payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_022_stage_order(input_data):
    # Task: stage order records and prepare transport-ready payload.
    task_name = 'stage_order'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'order',
        'stage': 'stage',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # ESCAPE: payload is appended to RETAINED_AUDIT, outliving function scope.
    RETAINED_AUDIT.append(payload)
    return 'ok'

File: tests/nodejs/cases/case_022_stage_order.js
/**
 * Task case 022: stage_order
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case022StageOrder(input) {
  // Task: stage order records and prepare transport-ready payload.
  const taskName = 'stage_order';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'order',
    stage: 'stage',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: payload is appended to retained audit state, outliving function scope.
  retainedAudit.push(payload);
  return 'ok';
}

module.exports = {
  case022StageOrder,
};
